### PR TITLE
Errors when config-merge

### PR DESCRIPTION
**Pull request by alex-schneider**
Multiple definitions of backend (eg. in proxy block) are not allowed.
Fix rewriting of backend blocks and error_file in error_handler.

---
<details>
    <summary>Reviewer checklist</summary>
    <ul>
        <li>Read PR description: a summary about the changes is required</li>
        <li>Changelog updated</li>
        <li>Documentation: docs/{Reference, Cli, ...}, Docker and cli help/usage</li>
        <li>Pulled branch, manually tested</li>
        <li>Verified requirements are met</li>
        <li>Reviewed the code</li>
        <li>Reviewed the related tests</li>
    </ul>
</details>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ Unreleased changes are available as `avenga/couper:edge` container.
   * Multiple labels for [`error_handler` blocks](./docs/ERRORS.md#error_handler-specification) ([#462](https://github.com/avenga/couper/pull/462))
   * [`error_handler` blocks](./docs/REFERENCE.md#error-handler-block) for an error type defined in both `endpoint` and `api` ([#469](https://github.com/avenga/couper/pull/469))
   * Request methods are treated case-insensitively when comparing them to methods in the `allowed_methods` attribute of [`api`](./docs/REFERENCE.md#api-block) or [`endpoint`](./docs/REFERENCE.md#endpoint-block) blocks ([#478](https://github.com/avenga/couper/pull/478))
+  * Rewriting of relative pathes when [merging](./docs/MERGE.md) configuration files ([#483](https://github.com/avenga/couper/pull/483))
```
Comment (malud) [inline CHANGELOG.md:27]: Merge Feature was added in this release, fixes should not be listed here.

File: config/configload/merge.go [hunk 2]
```diff
@@ -57,6 +62,52 @@ func absBackendBlock(backendBlock *hclsyntax.Block) {
...
+			}
+		}
+	}
+
+	return nil
+}
+
+// newErrorHandlerKey returns a merge key based on a possible mixed error-kind format.
+// "label1" and/or "label2 label3" results in "label1 label2 label3".
+func newErrorHandlerKey(block *hclsyntax.Block) (key string) {
```
Comment (malud) [inline config/configload/merge.go:97]: Reason for moving this method to top of the file? Helper methods should be not at first place due to readability reasons.

File: config/configload/merge.go [hunk 3]
```diff
@@ -13,13 +13,18 @@ import (
 	"github.com/zclconf/go-cty/cty"
 )
 
-func errorUniqueLabels(block *hclsyntax.Block) error {
+const (
+	errMultipleBackends = "Multiple definitions of backend are not allowed."
+	errUniqueLabels     = "All %s blocks must have unique labels."
+)
+
+func createError(msg string, block *hclsyntax.Block) error {
```
Comment (malud) [inline config/configload/merge.go:21]: golang pattern would be `newError` even better `newMergeError`

File: CHANGELOG.md (hunk 1 above)
Comment (malud) [reply CHANGELOG.md:27]: Was there a recent change related to this point?

**Review by malud — CHANGES_REQUESTED** [review]

File: config/configload/merge.go (hunk 2 above)
Comment (alex-schneider) [reply config/configload/merge.go:97]: All helper methods are currently placed in the top of this file. We can move all helper methods to the bottom of file.

File: CHANGELOG.md (hunk 1 above)
Comment (alex-schneider) [reply CHANGELOG.md]: is removed...

File: config/configload/merge.go (hunk 3 above)
Comment (alex-schneider) [reply config/configload/merge.go]: is renamed

File: config/configload/merge.go [hunk 4]
```diff
@@ -57,6 +62,52 @@ func absBackendBlock(backendBlock *hclsyntax.Block) {
 	}
 }
 
+func absInBackends(block *hclsyntax.Block) error {
```
[the remark this answers is not in the document]
Comment (alex-schneider) [reply config/configload/merge.go:65]: no, we have to count blocks **and** to search for backends in block-attributes. i will add a description...

**Review by malud — APPROVED** [review]

**Pull request merged**